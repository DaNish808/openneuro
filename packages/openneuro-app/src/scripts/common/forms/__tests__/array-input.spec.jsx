import React from 'react'
import { shallow } from 'enzyme'
import ArrayInput from '../array-input'

const emptyModel = []
const emptyValue = []

const hiddenCheckboxModel = [
  {
    id: 'license-key',
    type: 'checkbox',
  },
]
const hiddenCheckboxValue = ['false']

describe('common/forms/ArrayInput', () => {
  it('renders successfully', () => {
    const wrapper = shallow(
      <ArrayInput model={emptyModel} value={emptyValue} />,
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('renders a hidden checkbox', () => {
    const wrapper = shallow(
      <ArrayInput model={hiddenCheckboxModel} value={hiddenCheckboxValue} />,
    )
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('button > span').exists()).toBe(true)
    expect(wrapper.find('button > span').text()).toBe(' Hidden')
  })
})
