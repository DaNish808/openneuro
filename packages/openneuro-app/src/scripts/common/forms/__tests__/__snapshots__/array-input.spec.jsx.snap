// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/forms/ArrayInput renders a hidden checkbox 1`] = `
<div
  className="cte-edit-array"
>
  <div
    className="cte-array-items"
  >
    <ArrayItem
      index={0}
      item="false"
      key="0"
      model={
        Array [
          Object {
            "id": "license-key",
            "type": "checkbox",
          },
        ]
      }
      onEdit={[Function]}
      remove={[Function]}
    />
  </div>
  <div
    className="text-danger"
  />
  <div
    className="text-info"
  />
  <div
    className="form-inline"
  >
    <span>
      <div
        className="form-group float-label-input"
        key="license-key"
      >
        <button
          className="admin-button"
          key="license-key"
          onClick={[Function]}
        >
          <span>
            <i
              className="fa fa-square-o"
            />
             Hidden
          </span>
        </button>
      </div>
    </span>
    <br />
    <button
      className="cte-save-btn btn-admin-blue add-btn"
      onClick={[Function]}
    >
      add
    </button>
  </div>
</div>
`;

exports[`common/forms/ArrayInput renders successfully 1`] = `
<div
  className="cte-edit-array"
>
  <div
    className="text-danger"
  />
  <div
    className="text-info"
  />
  <div
    className="form-inline"
  >
    <span />
    <br />
    <button
      className="cte-save-btn btn-admin-blue add-btn"
      onClick={[Function]}
    >
      add
    </button>
  </div>
</div>
`;
